source/common/upstream/health_checker_base_impl.cc

```diff
@@ -291,6 +292,11 @@ HealthTransition HealthCheckerImplBase::ActiveHealthCheckSession::setUnhealthy(
     }
   }
 
+  if (host_->healthFlagGet(Host::HealthFlag::PENDING_ACTIVE_HC)) {
+    host_->healthFlagClear(Host::HealthFlag::PENDING_ACTIVE_HC);
+    changed_state = HealthTransition::Changed;
```

Comment:
Why do we consider this a changed transition? Comment?

Reply:
I did this so that we'll start including this host if it gets health checked but fails the first check. If we report `Unchanged` here then `reloadHealthyHosts` won't be called on the cluster and we'll continue to ignore the new hosts. I'll add a comment to this effect.